front/app/component-library/components/CheckboxWithLabel/index.tsx

```diff
@@ -54,12 +56,18 @@ const CheckboxWithLabel = ({
         size={size}
         usePrimaryBorder={usePrimaryBorder}
         name={name}
+        aria-label={ariaLabel}
         mr="8px"
       />
       <Box as="span" mr="4px">
         {label}
       </Box>
-      {labelTooltipText && <IconTooltip content={labelTooltipText} />}
+      {labelTooltipText && (
+        <IconTooltip
+          content={labelTooltipText}
+          role={ariaLabel ? 'none' : undefined}
+        />
+      )}
```

Comment:
If an ariaLabel is provided, we attach that to the checkbox instead, so we don't need to present the IconTooltip to screenreaders.